kratos/includes/kratos_parameters.h

```diff
@@ -168,7 +161,7 @@ class KRATOS_API(KRATOS_CORE) Parameters
 
         std::size_t mDistance = 0;                       /// The iterator distance
         nlohmann::json& mrValue;                         /// The original container
-        std::unique_ptr<Parameters> mpParameters;        /// The unique pointer to the base Parameter
+        Kratos::shared_ptr<nlohmann::json> mpRoot;
```

Comment:
huuuum, @RiccardoRossi

Reply:
the original stores a `unique_ptr` to `Parameters`, which in turn [stores a `shared_ptr` to `nlohmann_json`](https://github.com/KratosMultiphysics/Kratos/blob/3316a7b94c5f58d340d708e39ecc5e9b18197467/kratos/includes/kratos_parameters.h#L942) so it's **one less** heap allocations, if you're worrying about the heap.